interactive/dates.py

```diff
@@ -0,0 +1,19 @@
+from calendar import WEDNESDAY
+from datetime import date, timedelta
+
+
+def date_of_last_extract():
```

Comment:
I wrote a set of tests for this function to sanity check this logic. You might be able to reduce them or refactor, but it will be worth keeping some of them

```
def test_date_of_last_extract_sun_to_previous_wed(freezer):
    freezer.move_to("2022-05-15")
    assert date_of_last_extract() == date(2022, 5, 4)


def test_date_of_last_extract_mon_to_previous_wed(freezer):
    freezer.move_to("2022-05-16")
    assert date_of_last_extract() == date(2022, 5, 4)


def test_date_of_last_extract_tues_to_previous_wed(freezer):
    freezer.move_to("2022-05-17")
    assert date_of_last_extract() == date(2022, 5, 11)


def test_date_of_last_extract_wed_to_previous_wed(freezer):
    freezer.move_to("2022-05-18")
    assert date_of_last_extract() == date(2022, 5, 11)


def test_date_of_last_extract_thurs_to_previous_wed(freezer):
    freezer.move_to("2022-05-19")
    assert date_of_last_extract() == date(2022, 5, 11)
```

Reply:
Oops, I forgot to pull those over, fixed 👍